pkg/clustersync/clusterSync.go

```diff
@@ -282,12 +284,17 @@ func processClusterDelete(ctx context.Context, obj interface{}) {
 			clusterName)
 
 	case "ManagedClusterAddOn":
+		clusterName = obj.(*unstructured.Unstructured).GetNamespace() // Namespace reflects the name of the cluster
 		// When ManagedClusterAddOn (MCA) is deleted, search is disabled in the cluster. So, we delete the resources
 		// and edges for that cluster from db. But the cluster node is kept until MC is deleted.
-		deleteClusterNode = false
-		klog.V(3).Infof("Received delete for %s. Deleting Cluster resources and edges for cluster %s from the DB", kind,
-			clusterName)
-
+		if name == "search-collector" { // process only search-collector addon deletes
+			deleteClusterNode = false
+			klog.V(3).Infof("Received delete for %s. Deleting Cluster resources and edges for cluster %s from the DB", kind,
+				clusterName)
+		} else {
```

Comment:
This works, but it's more efficient to filter the resources with a FieldSelector at the informer.

Can you try this:
```go
filter := metav1.ListOptions{ FieldSelector: "metadata.name=search-collector" }
filteredDynamicFactory := dynamicinformer. NewFilteredDynamicSharedInformerFactory(dynamicClient,
		time.Duration(config.Cfg.RediscoverRateMS)*time.Millisecond, metav1.NamespaceAll, filter)

managedClusterAddonInformer := filteredDynamicFactory.ForResource(*managedClusterAddonGvr).Informer()
```

API reference: https://pkg.go.dev/k8s.io/client-go@v0.23.5/dynamic/dynamicinformer#NewFilteredDynamicSharedInformerFactory

Reply:
Added.

Reply:
Thanks!